Index: app/src/test/java/com/stolz/placessearch/details/DetailViewModelTest.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/test/java/com/stolz/placessearch/details/DetailViewModelTest.kt	(date 1576987728000)
+++ app/src/test/java/com/stolz/placessearch/details/DetailViewModelTest.kt	(date 1576987728000)
@@ -0,0 +1,38 @@
+package com.stolz.placessearch.details
+
+import androidx.databinding.BindingAdapter
+import com.stolz.placessearch.network.GoogleMapsApiService
+import okhttp3.ResponseBody
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers.anyInt
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+
+class DetailViewModelTest {
+
+    @Mock
+    lateinit var googleMapsApiService: GoogleMapsApiService
+
+    private val bindingAdapter = BindingAdapter
+
+    @Before
+    internal fun setUp() {
+        MockitoAnnotations.initMocks(this)
+    }
+
+    @Test
+    fun test_getStaticMap() {
+        val expectedResponse = ResponseBod
+        Mockito.`when`(
+            googleMapsApiService.getStaticMap(
+                centerMarker = anyString(),
+                placeMarker = anyString(),
+                zoom = anyInt(),
+                size = anyString()
+            )
+        ).thenReturn(retrofit2.mock.Calls.response())
+    }
+}
\ No newline at end of file
