Index: app/src/test/java/com/stolz/placessearch/details/DetailViewModelTest.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/test/java/com/stolz/placessearch/details/DetailViewModelTest.kt	(date 1577031975000)
+++ app/src/test/java/com/stolz/placessearch/details/DetailViewModelTest.kt	(date 1577031975000)
@@ -0,0 +1,59 @@
+package com.stolz.placessearch.details
+
+import android.content.Context
+import android.graphics.Bitmap
+import com.google.android.gms.maps.model.LatLng
+import com.stolz.placessearch.database.PlaceDao
+import com.stolz.placessearch.model.Place
+import junit.framework.Assert.assertEquals
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.setMain
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+
+class DetailViewModelTest {
+
+    @Mock
+    lateinit var mockContext: Context
+    @Mock
+    lateinit var mockFavoriteDatabase: PlaceDao
+    @Mock
+    lateinit var mockDetailRepository: DetailRepository
+    @Mock
+    lateinit var mockBitmap: Bitmap
+
+    private lateinit var detailViewModel: DetailViewModel
+    private val mainThreadSurrogate = newSingleThreadContext("test thread")
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        Dispatchers.setMain(Dispatchers.Unconfined)
+
+        detailViewModel = DetailViewModel(
+            mockContext,
+            mockFavoriteDatabase,
+            mockDetailRepository
+        )
+    }
+
+    @Test
+    fun test_getStaticMap() {
+        val testPlace = Place(id = "1", name = "testPlace", location = LatLng(1.0, 1.0))
+        runBlocking {
+            Mockito.`when`(mockDetailRepository.fetchStaticMap(mockContext, testPlace))
+                .thenReturn(mockBitmap)
+
+            detailViewModel.getStaticMap(testPlace)
+        }
+        assertEquals(mockBitmap, detailViewModel.staticMap.value)
+
+    }
+
+
+}
\ No newline at end of file
